File: 조준장/level3/징검다리 건너기.py
def solution(stones, k):
    answer = 0
    start = 1
    end = max(stones)
    
    while start <= end:
        cnt = 0
        mid = (start+end) // 2
        
        for stone in stones:
            if (stone - mid) <= 0:
                cnt += 1
            else:
                cnt = 0
                
            if cnt >= k:
                break
                
        if cnt >= k:
            answer = mid
            end = mid-1
            
        else:
            start = mid+1
                
        
    
    return answer

File: 이훈일/Programmers/Level3/징검다리 건너기.java
class Solution {
    public int solution(int[] stones, int k) {
        int min = stones[0];
        int max = stones[0];
        for(int i = 1; i < stones.length; i++) {
            min = Math.min(min, stones[i]);
            max = Math.max(max, stones[i]);
        }
        
        while(min < max) {
            int mid = (min + max + 1)/2;
            if(isPossible(mid, stones, k)) {
                min = mid;
            }else {
                max = mid - 1;
            }
        }
        
        return max;
    }
    
    public Boolean isPossible(int mid, int[] stones, int k) {
        int count = 0;
        for(int stone : stones) {
            if(stone < mid) {
                count++;
            }else {
                count = 0;
            }
            if(count == k) {
                return false;
            }
        }
        return true;
    }
}
